### PR TITLE
Update `dircmp` for 3.13

### PR DESCRIPTION
**Pull request by max-muoto**
Add `shallow` argument: https://github.com/python/cpython/pull/109499

### CODE REVIEW

**Review by JelleZijlstra — APPROVED** [review]
This looks right but we may want to make this param keyword-only: https://github.com/python/cpython/issues/57141#issuecomment-2227100514

File: stdlib/filecmp.pyi [hunk 1]
```diff
@@ -17,13 +17,23 @@ def cmpfiles(
...
-        hide: Sequence[AnyStr] | None = None,
-    ) -> None: ...
+    if sys.version_info >= (3, 13):
+        def __init__(
+            self,
+            a: GenericPath[AnyStr],
+            b: GenericPath[AnyStr],
+            ignore: Sequence[AnyStr] | None = None,
+            hide: Sequence[AnyStr] | None = None,
+            shallow: bool = True,
```
Comment (JelleZijlstra) [inline stdlib/filecmp.pyi:27]:
```suggestion
            *,
            shallow: bool = True,
```
My change to make it keyword-only should be in the latest beta

Comment (JelleZijlstra) [reply stdlib/filecmp.pyi:28]: Seems like GitHub doesn't have beta 4 yet...

**Comment by max-muoto** [comment]
@JelleZijlstra We should be able to go ahead with merging this now.

**Review by hauntsaninja — APPROVED** [review]

**Pull request merged**